Shared/mods/deathmatch/logic/lua/CLuaFunctionParser.h

```diff
@@ -541,6 +549,19 @@ struct CLuaFunctionParserBase
             }
             return static_cast<T>(result);
         }
+        else if constexpr (std::is_same_v<T, CLuaArgument>)
+        {
+            int iType = lua_type(L, index);
```

Comment:
This type check isn't required as we validate that using `TypeMatch` before using `PopUnsafe` .